### PR TITLE
[autopilot] Use local channel balance in allocation calculation

### LINKED ISSUE #2608: Autopilot miscalculates the total funds
The autopilot agent calculates the total funds available by adding the on chain wallet funds to the total channel allocation.

However, the channel allocation is erroneously calculated by adding the channel capacities instead of the local balance:
https://github.com/lightningnetwork/lnd/blob/a0c0e8edc9a58e8833fd32119964207205f9c764/autopilot/agent_constraints.go#L103

As the threshold is checked against this value, it can lead to unexpected results, especially as remote funding grows larger (which should be excluded from the calculation of the total funds).

Apparently there was a provision for this amount to be calculated properly, but the corresponding field (FundedAmt) is not populated accordingly:
https://github.com/lightningnetwork/lnd/blob/e2285732e47bacd74206f0bbf2c42c6ad8825548/autopilot/graph.go#L103

### PR DESCRIPTION
**Pull request by halseth**
Fixes #2608

### CODE REVIEW

File: autopilot/interface.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ type Node interface {
 // Channel is a simple struct which contains relevant details of a particular
 // channel within the channel graph. The fields in this struct may be used a
 // signals for various AttachmentHeuristic implementations.
-type Channel struct {
+type LocalChannel struct {
```
Comment (bhandras) [inline autopilot/interface.go:42]: nit: godoc needs an update too.

File: pilot.go [hunk 2]
```diff
@@ -260,16 +260,35 @@ func initAutoPilot(svr *server, cfg *lncfg.AutoPilot,
...
+					Balance: balance,
 					Node: autopilot.NewNodeID(
 						channel.IdentityPub),
 				}
 			}
 
 			return chanState, nil
 		},
+		ChannelInfo: func(chanPoint wire.OutPoint) (
+			autopilot.LocalChannel, error) {
```
Comment (bhandras) [inline pilot.go:277]: why not return `*autopilot.LocalChannel`, that way returning an error would be cleaner (`nil` return value)?

File: pilot.go [hunk 3]
```diff
@@ -260,16 +260,35 @@ func initAutoPilot(svr *server, cfg *lncfg.AutoPilot,
...
+			channel, err := svr.remoteChanDB.FetchChannel(chanPoint)
+			if err != nil {
+				return autopilot.LocalChannel{}, err
+			}
+
+			localCommit := channel.LocalCommitment
+			return autopilot.LocalChannel{
+				ChanID:  channel.ShortChanID(),
+				Balance: localCommit.LocalBalance.ToSatoshis(),
+				Node: autopilot.NewNodeID(
```
Comment (bhandras) [inline pilot.go:288]: nit: no need for line break

File: autopilot/interface.go [hunk 4]
```diff
@@ -211,14 +211,4 @@ type ChannelController interface {
 	//
 	// TODO(roasbeef): add force option?
 	CloseChannel(chanPoint *wire.OutPoint) error
-
-	// SpliceIn attempts to add additional funds to the target channel via
-	// a splice in mechanism. The new channel with an updated capacity
-	// should be returned.
-	SpliceIn(chanPoint *wire.OutPoint, amt btcutil.Amount) (*Channel, error)
```
Comment (cfromknecht) [inline autopilot/interface.go:224]: Just a taddd premature 😂

**Review by cfromknecht — APPROVED** [review]
Nice diff, LGTM  😎

File: pilot.go (hunk 2 above)
Comment (halseth) [reply pilot.go]: It was just to not have to dereference the returned value before passing it to `OnChannelEdge`, but changed now: https://github.com/lightningnetwork/lnd/pull/4666/commits/2a3114641e27db7d640a0d8dd09ccb3cdc974710

File: pilot.go (hunk 3 above)
Comment (halseth) [reply pilot.go]: Fixed: https://github.com/lightningnetwork/lnd/pull/4666/commits/92dcd47f267c1d2cd19523be37bb36d6bba4fc33

File: autopilot/interface.go (hunk 1 above)
Comment (halseth) [reply autopilot/interface.go:42]: Fixed: https://github.com/lightningnetwork/lnd/pull/4666/commits/d3ccbd174ac53ca670c84c8c8a9298c8d5457b83

File: autopilot/interface.go (hunk 4 above)
Comment (halseth) [reply autopilot/interface.go:224]: I could also implement it in this PR if that is preferred?

Comment (bhandras) [reply autopilot/interface.go:224]: Please :slightly_smiling_face:

File: autopilot/interface.go (hunk 1 above)
Comment (bhandras) [reply autopilot/interface.go:42]: nit: s/may be used a/may be used as

**Review by bhandras — APPROVED** [review]
LGTM :1st_place_medal:

**Pull request merged**